src/GraphClient.ts

```diff
@@ -50,25 +55,68 @@ export default class GraphClient {
     // GraphQL query that is parsed into the standard GraphQL AST(Abstract syntax tree)
     const gqlSubscriptionQry = gql`${subscriptionQry}`;
     // Subscription handling
-    const querySubscriber = await Promise.resolve(this.apolloClient.subscribe({
+    const observable = this.apolloClient.subscribe({
       query: gqlSubscriptionQry,
       variables: {},
-    }).subscribe({
-      async next(response) {
-        const transactions = await fetcher.fetch(response.data);
-        await handler.handle(transactions);
-        // Integrate updation of ContractEntity uts here. Handlers should make sure error
-        // Promise.reject is thrown on error cases.
-      },
-      error(err) {
-        // Log error using logger
-        Logger.error(err);
-      },
-    }));
+    });
+    const querySubscriber = await Promise.resolve(
+      observable
+        .subscribe({
+          async next(response) {
+            const transactions = await fetcher.fetch(response.data);
+            await handler.handle(transactions);
+            await GraphClient.updateLatestUTS(
+              transactions,
+              response.data,
+              contractEntityRepository,
+            );
+          },
+          error(err) {
+            Logger.error(err);
+          },
+        }),
+    );
 
     return querySubscriber;
   }
 
+  /**
+   * This method updates latest timestamp for contract entities.
+   * @param transactions Transactions for transaction fetcher.
+   * @param subscriptionResponse Subscription response.
+   * @param contractEntityRepository Instance of contract entity repository.
+   */
+  private static async updateLatestUTS(
+    transactions: Record<string, Record<string, any>>,
+    subscriptionResponse: Record<string, any[]>,
+    contractEntityRepository: ContractEntityRepository,
+  ): Promise<void> {
+    const savePromises = Object.keys(transactions).map(
+      async (transactionKind) => {
+        const { contractAddress } = subscriptionResponse[transactionKind][0];
+        const transaction = transactions[transactionKind].length > 0
+          ? transactions[transactionKind][transactions[transactionKind].length - 1]
+          : null;
+
+        const number = transaction ? transaction.uts : 0;
+        const currentUTS = new BigNumber(
+          number,
+        );
+
+        const contractEntity = new ContractEntity(
+          contractAddress,
+          transactionKind,
+          currentUTS,
+        );
+        return contractEntityRepository.save(
+          contractEntity,
+        );
+      },
+    );
+
+    await Promise.all(savePromises);
```

Comment:
Promise.all calls promises in parallel. Do we need to save entries in some order, sequential, or (as those are UTC values)?

We should assure, that we do not update utc to smaller value in ContractEntityRepository. Thoughts?

Another thought, should transaction handler(s) update ContractEntityRepository on successful handle? Thoughts?

Reply:
It will only save for one entity with the highest uts. The query sorts the record from increasing order.

Reply:
On L102 selects last transaction of a kind. 
```
transactions[transactionKind][transactions[transactionKind].length - 1]
```

If you see queries in `SubscriptionQueries` class, transactions from an entity are sorted in increasing order of timestamp. If we select the last record, it will always ensure the latest timestamp for that transaction kind.

Secondly, We iterate on different transaction kind and select the last transaction. 
which means `savePromises` will have only one promise for each entity type/transaction kind with the latest timestamp/uts.

Reply:
Transaction handler only knows about how to handle transactions. This is the responsibility of the subscription module to maintain timestamp /uts. I feel transaction handler should be incognito of this timestamp/uts logic.